processing/src/main/java/io/druid/segment/QueryableIndex.java

```diff
@@ -31,17 +32,19 @@
  */
 public interface QueryableIndex extends ColumnSelector, Closeable
 {
-  public Interval getDataInterval();
-  public int getNumRows();
-  public Indexed<String> getAvailableDimensions();
-  public BitmapFactory getBitmapFactoryForDimensions();
-  public Metadata getMetadata();
-  public Map<String, DimensionHandler> getDimensionHandlers();
+  Interval getDataInterval();
+  int getNumRows();
+  Indexed<String> getAvailableDimensions();
+  BitmapFactory getBitmapFactoryForDimensions();
+  Metadata getMetadata();
+  Map<String, DimensionHandler> getDimensionHandlers();
+
+  SearchQueryDecisionHelper getDecisionHelper();
```

Comment:
IMO this decision-making should be self-contained within the Search query rather than bleeding into the QueryableIndex interface.

My rationale is that the thresholds embodied in this decision helper are experimentally determined and specific to the Search query algorithm, and ideally nothing else should know about them. Moving this to the Search query does mean that the Search query will need to know about different kinds of bitmaps, but I think that's better than the QueryableIndex knowing about the Search query.